src/codechandler/CodecHandler.cpp

```diff
@@ -0,0 +1,48 @@
+#include "CodecHandler.h"
+
+bool CodecHandler::GetInformation(kodi::addon::InputstreamInfo& info)
+{
+  AP4_GenericAudioSampleDescription* asd(nullptr);
+  if (m_sampleDescription)
+  {
+    if ((asd = dynamic_cast<AP4_GenericAudioSampleDescription*>(m_sampleDescription)))
+    {
+      if ((!info.GetChannels() && asd->GetChannelCount() != info.GetChannels()) ||
+          (!info.GetSampleRate() && asd->GetSampleRate() != info.GetSampleRate()) ||
+          (!info.GetBitsPerSample() && asd->GetSampleSize() != info.GetBitsPerSample()))
+      {
+        if (!info.GetChannels())
+          info.SetChannels(asd->GetChannelCount());
+        if (!info.GetSampleRate())
+          info.SetSampleRate(asd->GetSampleRate());
+        if (!info.GetBitsPerSample())
+          info.SetBitsPerSample(asd->GetSampleSize());
+        return true;
```

Comment:
to replace integer value's handled as booleans
e.g.:
`if ((!info.GetChannels() && asd->GetChannelCount() != info.GetChannels()) ||`
to
`if ((info.GetChannels() == 0 && asd->GetChannelCount() != info.GetChannels()) ||`